testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergHiveTablesCompatibility.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.iceberg;
+
+import io.trino.tempto.ProductTest;
+import org.testng.annotations.Test;
+
+import static io.trino.tempto.assertions.QueryAssert.assertQueryFailure;
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.tests.product.TestGroups.ICEBERG;
+import static io.trino.tests.product.TestGroups.STORAGE_FORMATS;
+import static io.trino.tests.product.hive.util.TemporaryHiveTable.randomTableSuffix;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+
+public class TestIcebergHiveTablesCompatibility
+        extends ProductTest
+{
+    @Test(groups = {ICEBERG, STORAGE_FORMATS})
+    public void testIcebergSelectFromHiveTable()
+    {
+        String tableName = "test_iceberg_select_from_hive_" + randomTableSuffix();
+        onTrino().executeQuery("CREATE TABLE hive.default." + tableName + "(a bigint)");
+
+        assertQueryFailure(() -> onTrino().executeQuery("SELECT * FROM iceberg.default." + tableName))
+                .hasMessageMatching("Query failed \\(#\\w+\\):\\Q Not an Iceberg table: default." + tableName);
+
+        assertQueryFailure(() -> onTrino().executeQuery("SELECT * FROM iceberg.default.\"" + tableName + "$files\""))
+                .hasMessageMatching("Query failed \\(#\\w+\\):\\Q Wrong table type: FILES");
```

Comment:
Shouldn't this error message be different too? "Not an Iceberg table" maybe?

Reply:
going to be fixed in https://github.com/trinodb/trino/pull/8692, will add a TODO for now

Reply:
added TODO linking to https://github.com/trinodb/trino/issues/8690